qiime2/core/cache.py

```diff
@@ -754,6 +760,18 @@ def garbage_collection(self):
                     set_permissions(target, None, USER_GROUP_RWX)
                     shutil.rmtree(target)
 
+    def _check_dangling_reference(self, data_path, key_path):
+        """ If the data specified does not exist then remove the key
+        """
+        if not os.path.exists(data_path):
+            warnings.warn(f"Dangling reference {key_path}. Data at {data_path}"
+                          " does not exist. Reference will be removed")
+            os.path.remove(data_path)
```

Comment:
```suggestion
            os.path.remove(key_path)
```

(I assume)

Reply:
Whoops yeah good catch. Probably made that while moving things into that helper